### ISSUE TITLE
1.16.4: Item Variable

### ISSUE THREAD

**Issue by Ash10744** [kangarko/ChatControl#418]
Minecraft Version: 1.16.4
MySQL: true
BungeeCord: true

The [item] variable works for anyone who is an operator or server admin but when a player decides to do the variable it restarts the server.

Error Log:
https://mcpaste.io/13bc05e990f351c1

My Variable config:
https://mcpaste.io/bc8268e17f1e0329

**Comment by ElBananaa** [comment]
Hey!

Please provide us your `/chc debug` file!
Also make sure your plugins are all up-to-date.

**Comment by Ash10744** [comment]
https://ufile.io/z0jwp5yy

**Comment by ElBananaa** [comment]
Alright, replace your actual item.yml file content with this: https://hastebin.com/uqojizokug.php and your i.yml file content with this https://hastebin.com/tulumivine.php
Let me know if it fixed your issue.

Also, don't forget to update your plugins. In more than 70% of the cases, issues are related to an outdated version, and as I can see, a lot of your plugins aren't up-to-date (And not only chatcontrol and its dependencies)

**Comment by Ash10744** [comment]
Yep all fixed thank you!

**Comment by Ash10744** [comment]
Right its still occuring I think whenever there is custom colours Like 1.16 HEX Colours it crashes the server.

**Comment by Ash10744** [comment]
Yeah the custom colours on  1.16 are causing an issue it crashes the server.

**Comment by Ash10744** [comment]
And any 1.16 items also crash the server.

**Comment by ElBananaa** [comment]
> Yeah the custom colours on 1.16 are causing an issue it crashes the server.

I just tried on my own server with the latest version, and it works perfectly fine, even with hex colors. It could be caused by the item lore, or by another plugin tho. Try to reproduce the issue on a local server with only chatcontrol-red installed, and let me know if it still crashes.

**Comment by Ash10744** [comment]
I will update my plugin quickly then try and do it.

**Comment by Ash10744** [comment]
Right it is still crashing the server

**Comment by Ash10744** [comment]
This is one of the items thats crashing the server when doing it:
![image](https://user-images.githubusercontent.com/16086663/103319386-980c7580-4a29-11eb-8d93-61f84daf0e80.png)

**Comment by ElBananaa** [comment]
I think the only way to actually fix this issue (for now) would be to replace the `Hover_Item: player.getInventory().getItemInMainHand()` with `Hover_Item: []`
So the item lore won't be shown.

**Comment by Ash10744** [comment]
I can do that temporarily but I really need to show it in chat for the entire purpose of my server?

**Comment by ElBananaa** [comment]
> I can do that temporarily but I really need to show it in chat for the entire purpose of my server?

Hey!

Please use this line for now. I don't know if there's anything that could be done tho.
Please wait for Kangarko to see if there's anything he could do to fix that.

**Comment by kangarko** [comment]
Please post me the full crash log.

**Comment by Ash10744** [comment]
It's weird cause there is no particular error showing when it disconnects someone.


https://mcpaste.io/4f8efb33f424e97b

**Comment by Ash10744** [comment]
After investigating it further its kicking certain people... Like anyone who can see the item in chat gets kicked but it shows the server restarting for the individual player which is so weird...

**Comment by kangarko** [comment]
Try setting Integration.ProtocolLib.Listen_For_Packets and Tab_Complete.Enabled both to false in settings.yml and see if the kicks stop

**Comment by Ash10744** [comment]
Sorry for the late reply I have had issues irl.


This is still occurring even on the latest version. Just tried with this item:
![image](https://user-images.githubusercontent.com/16086663/103847034-4bfba980-5097-11eb-8789-513eddb98334.png)

**Comment by Ash10744** [comment]
Maybe it could be related to the new Unicode introduced to 1.16?

**Comment by kangarko** [comment]
When the player gets kicked what do they see on screen?

**Comment by Ash10744** [comment]
It shows restarting the server but only for that player

**Comment by Ash10744** [comment]
Or the people who have the view permission of the variable

**Comment by kangarko** [comment]
Can you upload a screenshot of what they see on their screen? Really strange

**Comment by Ash10744** [comment]
A really odd thing relating to this I spent a couple of hours looking through my plugins that could have caused this. Then I noticed that my auto-reconnect plugin in bungee was actually causing the issue relating to this. Which is super weird.


Here is the plugin here if you're interested further I am so sorry for wasting your time and realizing it was something completely different (which probably explained the lack of information it gave me regarding the crash)
https://www.spigotmc.org/resources/autoreconnect-bungeecord-reconnect.62729/
